### DISCUSSION TITLE
Only getting swup partial animation

### DISCUSSION THREAD

**Discussion by abstractTechnica** [swup/swup#413] [Help]
I'm storing my nav links in a separate html file and using a script to call it. when I click on a link to navigate to the next page, the animation starts to move but before it finishes, it loads the next page. I can make it work if I don't store the nav links in a separate file but there has to be a way to make this work so that I have a single html file for the menu.

**Comment by hirasso** [comment]
@abstractTechnica you can control the duration of your transitions using a *single* element with a class of `transition-[something]` on your page. Swup will look at this element's css property `transition-duration` and wait for it to finish before swapping to the new page's contents.
